### DISCUSSION TITLE
Not able to connect MinIO with railway.

### DISCUSSION THREAD

**Discussion by tusharmctrl** [minio/minio#18580] [Q&A]
Okay, help me out here guys!

- I have MinIO (Console and Bucket) deployed on railway. Everything is absolutely fine with Console and working like a charm.

My Bucket is  publicly exposed  and it is using this endpoint (https://bucket-production-cafb.up.railway.app)

![image](https://github.com/minio/minio/assets/141150938/267305eb-79cd-4da0-8a19-0216948d93c5)

The good thing is when I try to access object using the link such as - `https://bucket-production-cafb.up.railway.app/bucket-name/file-name`. It is completely accessible. That means we can say instance is correctly deployed?

In my Next JS application I am using `minio` npm package. And the definition goes something like this.

```
 const minio = new Minio.Client({
    endPoint: env.MINIO_ENDPOINT,
    useSSL: false,
    port: 9000,
    accessKey: env.MINIO_ACCESS_KEY,
    secretKey: env.MINIO_SECRET_KEY
  });
  ```
  
 ```MINIO_ENDPOINT=bucket-production-cafb.up.railway.app```
 
 And the code of uploading an object is something like this - 
 
 ```
 minio.putObject(bucket, objectName, file.buffer, file.size, { "Content-Type": file.mimetype }, (err, etag) => {
            if (err) {
              console.error("Error uploading object:", err);
              return res.status(500).json({ error: "Could not upload the file" });
            } else {
              const fileObject = {
                originalName: objectName,
                type: req.file?.mimetype,
                etag: etag
              };
              return res.status(200).json(fileObject);
            }
 ```
 
 The error log is here:
 
 ```
 Error uploading object: Error: connect ETIMEDOUT 34.87.74.158:9000
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1495:16) {
  errno: -4039,
  code: 'ETIMEDOUT',
  syscall: 'connect',
  address: '34.87.74.158',
  port: 9000
}
```

Error log from railway: 
```
%!(EXTRA *websocket.CloseError=websocket: close 1006 (abnormal closure): unexpected EOF)Error while reading objectManager message.
Error while reading objectManager message
```

Will apprecialte a little help. Thanks ((:

**Comment by EduardoProfe666** [comment]
Hi there... At this point I don't know if u still want an answer, but I had the same problem than u. In my case i'm using .Net Minio Sdk, but it was the same error. My solution was to add a PORT = 9000 variable to the bucket service. The console port was 9000 too. With this solution u can access both the console app and the S3 app through Minio Sdk... 
<img width="596" alt="bucket _ Railway - Google Chrome 12_02_2024 11_53_48" src="https://github.com/minio/minio/assets/119138695/e135cc24-0fa6-4c74-977d-e6d8a537f707">

**Reply by arthurdedeus** [reply]
This worked for me as well @EduardoProfe666! Thanks a lot for posting your solution here.
